src/core/fieldformatter/qgsvaluerelationfieldformatter.cpp

```diff
@@ -180,7 +180,14 @@ QStringList QgsValueRelationFieldFormatter::valueToStringList( const QVariant &v
     if ( newVal.toString().trimmed().startsWith( '{' ) )
     {
       newVal = QVariant( newVal.toString().trimmed().mid( 1 ).mid( 0, newVal.toString().length() - 2 ).prepend( '[' ).append( ']' ) );
+
+      if ( !json::accept( newVal.toString().toStdString() ) )
```

Comment:
what accept does? 
If you are trying to parse json, you probably want `parse()` within a try/catch block https://nlohmann.github.io/json/classnlohmann_1_1basic__json_afd4ef1ac8ad50a5894a9afebca69140a.html#afd4ef1ac8ad50a5894a9afebca69140a

Reply:
It's to check if the replacement of the `{}` resulted in a correct JSON format (means that the `parse` can be successful later). If not (because the data are still stored earlier without quotes), it decodes the hstore-list like in the old days.

Reply:
yeah, I guessed that it was your intention, the usage seems correct, I was mislead by another `accept`: https://nlohmann.github.io/json/classnlohmann_1_1basic__json_ae797958b922732bf5fc01053d0659c1f.html#ae797958b922732bf5fc01053d0659c1f